crates/autopilot/src/domain/competition/winner_selection/combinatorial.rs

```diff
@@ -350,3 +350,798 @@ type ScoreByDirection = HashMap<DirectedTokenPair, Score>;
+            ],
+            "auction": {
+                "orders": {
+                    "Order 1": ["sell", "Token A", amount(1_000), "Token B", amount(1_000)],
+                    "Order 2": ["sell", "Token C", amount(1_000), "Token D", amount(1_000)],
+                    "Order 3": ["sell", "Token A", amount(1_000), "Token D", amount(1_000)]
+                }
+            },
+            "solutions": {
+                // best batch
+                "Solution 1": {
+                    "solver": "Solver 1",
+                    "trades": {
+                        "Order 1": [amount(1_000), amount(1_100)],
+                        "Order 2": [amount(1_000), amount(1_100)]
+                    },
+                    "score": score(200),
+                },
+                // compatible batch
+                "Solution 2": {
+                    "solver": "Solver 1", // same solver
+                    "trades": {
+                        "Order 3": [amount(1_000), amount(1_100)],
+                    },
+                    "score": score(100),
+                }
+            },
+            "expected_fair_solutions": ["Solution 1", "Solution 2"],
+            "expected_winners": ["Solution 1", "Solution 2"],
+            "expected_reference_scores": {
+                "Solver 1": "0",
+            },
+        });
+        TestCase::from_json(case).validate();
+    }
+
+    #[test]
+    // Incompatible bid does not win but increases the reference score of the winner
+    fn incompatible_bids() {
+        let case = json!({
+            "tokens": [
+                ["Token A", address(0)],
+                ["Token B", address(1)],
+                ["Token C", address(2)],
+                ["Token D", address(3)]
+            ],
+            "auction": {
+                "orders": {
+                    "Order 1": ["sell", "Token A", amount(1_000), "Token B", amount(1_000)],
+                    "Order 2": ["sell", "Token C", amount(1_000), "Token D", amount(1_000)],
+                }
+            },
+            "solutions": {
+                // best batch
+                "Solution 1": {
+                    "solver": "Solver 1",
+                    "trades": {
+                        "Order 1": [amount(1_000), amount(1_100)],
+                        "Order 2": [amount(1_000), amount(1_100)]
+                    },
+                    "score": score(200),
+                },
+                // compatible batch
+                "Solution 2": {
+                    "solver": "Solver 2",
+                    "trades": {
+                        "Order 1": [amount(1_000), amount(1_100)],
+                    },
+                    "score": score(100),
+                }
+            },
+            "expected_fair_solutions": ["Solution 1", "Solution 2"],
+            "expected_winners": ["Solution 1"],
+            "expected_reference_scores": {
+                "Solver 1": "100",
+                "Solver 2": "200",
```

Comment:
We should only have a reference score for `Solver 1` since only that provided a winning solution.